### ISSUE TITLE
% (percent sign) showing up while using zsh

### ISSUE THREAD

**Issue by ajitid** [zeit/hyper#2144]
OS: Ubuntu 17.04

It happens only for first tab opened in hyper.

https://cloud.githubusercontent.com/assets/593151/26001911/35b10a48-372e-11e7-90c6-a11058362c7f.png

**Comment by albinekb** [comment]
I have a similar bug with fish!

First tab:
![image](https://user-images.githubusercontent.com/5027156/29922648-cb582ebc-8e56-11e7-8f51-bbdf9c072bc1.png)

Second tab (or after cmd+k):
![image](https://user-images.githubusercontent.com/5027156/29922657-d32c5ee2-8e56-11e7-932e-d6d9f4b20055.png)

**Comment by ppot** [comment]
This is du to the session shell invoking. Since we print the window before creating the required IPC connectors.

**Comment by ajitid** [comment]
Can't this be cleared at start for first tab?

**Comment by ppot** [comment]
Actually no. It buffer your input and emit it once it's ready. If you type before it's complete you will have the `%` sign.

**Comment by ajitid** [comment]
No, I am saying can Hyper detect that zsh is ready? Once it detects that, it clears the terminal.

**Comment by ajitid** [comment]
Don't know how but VS Code's terminal is able to handle it

**Comment by albinekb** [comment]
VS Code uses `xterm`, we use `hterm` as our terminal library, that's why VS Code is better in many ways, and we're in progress of migrating to `xterm` @ajitid

**Comment by infctr** [comment]
adding this `unsetopt PROMPT_SP` to my `.zshrc` helped  
[reference](https://superuser.com/questions/645599/why-is-a-percent-sign-appearing-before-each-prompt-on-zsh-in-windows)

**Comment by albinekb** [comment]
That's a good workaround, but I think we should fix the issue so that hyper handles the row correctly

**Comment by ajitid** [comment]
@albinekb I'm not using Hyper at present. Is this issue resolved in Hyper 2.0?

**Comment by jakubziemba** [comment]
The issue is back in version 3.0.0.

**Comment by Barben360** [comment]
The issue is back in version 3.0.0 but adding `unsetopt PROMPT_SP` seems to fix it.

**Comment by chris-c-thomas** [comment]
I've added `unsetopt PROMPT_SP` to my .zshrc and that fixed it. Is that the permanent solution or should this be fixed on Zeit's end?

**Comment by meatherly** [comment]
This helped me understand the problem. https://unix.stackexchange.com/a/167600

**Comment by ifyour** [comment]
version 3.0.2 same problem. 😟

```bash
# .zshrc
# fix Hyper first line precent sign

unsetopt PROMPT_SP
```

**Comment by mserajnik** [comment]
Can also confirm that it was fixed in `2.x`, but came back with `3.0.0`.

Using the workaround `unsetopt PROMPT_SP` for now.

**Comment by zachbryant** [comment]
> The issue is back in version 3.0.0 but adding `unsetopt PROMPT_SP` to .zshrc seems to fix it.

This is not a fix

**Comment by zachauten** [comment]
I'm currently seeing this issue in `3.0.2`. Interestingly it shows up in the vscode integrated terminal as well.

**Comment by mikeguidry** [comment]
I think its caused by an applications stdout piping without a finish line..
my setup:
gnome terminal, ubuntu 18,
ZSH_THEME="powerlevel9k/powerlevel9k"

....
 mike@i  ~  echo -n hi                                                                                                                                                      ✔  10003  07:44:30
hi%                                                                                                                                                                                                  mike@i  ~  

in other words.. doing echo -n text will print 'text%'   if you perform unset PROMPT_SP then you will never see the hi or text part

**Comment by rafael-paiva** [comment]
Why is this closed? Is there a fix (not workaround)?

**Comment by edsu** [comment]
Thanks for the `unsetopt PROMPT_SP` trick -- this suddenly became an issue for me as OS X pushed me off bash to zsh. I'd been meaning to try it for ages anyhow, and this was the nudge I needed I guess...

**Comment by Luxcium** [comment]
I have same issue when not using hyper (iTerm) but only when loading inside of TMUX I load my ZSH exactly the same way outside and inside the TMUX but I am getting the `%` only when in TMUX session ...

**Comment by Luxcium** [comment]
> I have the same issue when I am not using hyper (using iTerm) but only when loading inside of TMUX... I load my ZSH session exactly the same way outside and inside the TMUX but I am getting the `%` only when in TMUX session ...
> 
> on the image below the right-side displays, the `%` after loading ZSH directly from TMUX and the left was showing the same `%` but after invoking `ZSH` again manually it's not displayed
> <img alt="Capture d’écran, le 2020-01-23 à 03 30 57" width="1848" src="https://user-images.githubusercontent.com/42672814/72968143-e385c080-3d90-11ea-8d4b-4323ecf6b48c.png">

I should have read the comment before sending my comment I am sorry ~>

PROMPT_SP <D>
Attempt to preserve a partial line (i.e. a line that did not end with a newline) that would otherwise be covered up by the command prompt due to the PROMPT_CR option. This works by outputting some cursor-control characters, including a series of spaces, that should make the terminal wrap to the next line when a partial line is present (note that this is only successful if your terminal has automatic margins, which is typical).

When a partial line is preserved, by default you will see an inverse+bold character at the end of the partial line: a ‘%’ for a normal user or a ‘#’ for root. If set, the shell parameter PROMPT_EOL_MARK can be used to customize how the end of partial lines are shown.

NOTE: if the PROMPT_CR option is not set, enabling this option will have no effect. This option is on by default.

**Comment by fernandopasik** [comment]
It doesn't happen in the canary 3.1.0

**Comment by francogp** [comment]
I'm having this problem on the latest commits, since today. unsetopt PROMPT_SP does not fix the problem

**Comment by LabhanshAgrawal** [comment]
@francogp check the note by @Luxcium about PROMPT_CR, that might help.

**Comment by Liquidmantis** [comment]
I started seeing this recently, too.  I switched to the canary channel, and it doesn't show up, but as soon as I start typing I lose my prompt and all the terminal text is invisible, so I switched back to stable.

**Comment by iruma-kun** [comment]
another workaround is adding "clear" to zshrc, but might cause issues if u wanna see session restoration and welcome messeges.
